tools/path-encoding-verify.py

```
@@ -0,0 +1,116 @@
+    """Fail closed even under ``python -O`` (asserts would be stripped)."""
+    if not condition:
+        raise ValueError(f"{name}: {detail}")
+
+
+def reference_encode(key: str) -> str:
+    """cachekit-py ``CachekitIOBackend._encode_key`` (f000ba3)."""
+    encoded = quote(key, safe="")
+    return encoded.replace(".", "%2E") if encoded in (".", "..") else encoded
+
+
+def check_segment(name: str, segment: str, dot_segment: bool, key: str, pattern: re.Pattern[str] = SEGMENT) -> None:
```

Ruff (0.16.3)

[warning] Boolean-typed positional argument in function definition

(FBT001)